core/unit_test/TestTeam.hpp

```diff
@@ -1338,11 +1338,12 @@ struct TestTeamBroadcast<
   template <class ScalarType>
   static inline
       typename std::enable_if<!std::is_integral<ScalarType>::value, void>::type
-      compare_test(ScalarType A, ScalarType B) {
-    if (std::is_same<ScalarType, double>::value) {
-      ASSERT_DOUBLE_EQ((double)A, (double)B);
-    } else if (std::is_same<ScalarType, float>::value) {
-      ASSERT_FLOAT_EQ((double)A, (double)B);
+      compare_test(ScalarType A, ScalarType B, double epsilon_factor) {
+    if (std::is_same<ScalarType, double>::value ||
+        std::is_same<ScalarType, float>::value) {
```

Comment:
Any good reason not to use `std::is_floating_point`?

Reply:
I was thinking of the addition of half and the ambiguity of what is_floating_point means in that case ...

Reply:
> I was thinking of the addition of `half` and the ambiguity of what `is_floating_point` means in that case ...

It means `float`, `double`, and `long double`.  Other types are not "floating-point types" in C++ <= 20.  If one of those papers that proposes generalizing this notion lands, then you might like to ask `is_iec559` and write generic IEEE 754 - flavored bounds.